go/test/endtoend/vtgate/queries/random/random_test.go

```diff
@@ -0,0 +1,517 @@
+
+	// EOF (errno 2013) (sqlstate HY000)
+	helperTest(t, "select /*vt+ PLANNER=Gen4 */ count(*) from dept as tbl0, (select /*vt+ PLANNER=Gen4 */ count(*) from emp as tbl0, emp as tbl1 limit 18) as tbl1")
+
+	// unsupported
+	// unsupported: in scatter query: complex aggregate expression (errno 1235) (sqlstate 42000)
+	helperTest(t, "select /*vt+ PLANNER=Gen4 */ (select count(*) from emp as tbl0) from emp as tbl0")
+
+	// unsupported
+	// unsupported: using aggregation on top of a *planbuilder.orderedAggregate plan
+	helperTest(t, "select /*vt+ PLANNER=Gen4 */ count(*) from (select count(*) from dept as tbl0) as tbl0")
+
+	// unsupported
+	// unsupported: using aggregation on top of a *planbuilder.orderedAggregate plan
+	helperTest(t, "select /*vt+ PLANNER=Gen4 */ count(*), count(*) from (select count(*) from dept as tbl0) as tbl0, dept as tbl1")
+
+	// unsupported
+	// EOF (errno 2013) (sqlstate HY000)
+	helperTest(t, "select /*vt+ PLANNER=Gen4 */ count(*), count(*) from (select count(*) from dept as tbl0 group by tbl0.deptno) as tbl0")
+
+	// unsupported
+	// unsupported: in scatter query: aggregation function 'avg(tbl0.deptno)'
+	helperTest(t, "select /*vt+ PLANNER=Gen4 */ avg(tbl0.deptno) from dept as tbl0")
+}
+
+func TestRandom(t *testing.T) {
+	mcmp, closer := start(t)
+	defer closer()
+
+	require.NoError(t, utils.WaitForAuthoritative(t, keyspaceName, "emp", clusterInstance.VtgateProcess.ReadVSchema))
+	require.NoError(t, utils.WaitForAuthoritative(t, keyspaceName, "dept", clusterInstance.VtgateProcess.ReadVSchema))
+
+	// specify the schema (that is defined in schema.sql)
+	schemaTables := []tableT{
+		{Name: sqlparser.NewTableName("emp")},
+		{Name: sqlparser.NewTableName("dept")},
+	}
+	schemaTables[0].AddColumns([]column{
+		{Name: "empno", Typ: "bigint"},
+		{Name: "ename", Typ: "varchar"},
+		{Name: "job", Typ: "varchar"},
+		{Name: "mgr", Typ: "bigint"},
+		{Name: "hiredate", Typ: "date"},
+		{Name: "sal", Typ: "bigint"},
+		{Name: "comm", Typ: "bigint"},
+		{Name: "deptno", Typ: "bigint"},
+	}...)
+	schemaTables[1].AddColumns([]column{
+		{Name: "deptno", Typ: "bigint"},
+		{Name: "dname", Typ: "varchar"},
+		{Name: "loc", Typ: "varchar"},
+	}...)
+
+	endBy := time.Now().Add(10 * time.Second)
+
+	var queryCount int
+	for time.Now().Before(endBy) && (!t.Failed() || TestFailingQueries) {
+		query := sqlparser.String(randomQuery(schemaTables, 3, 3))
+		_, vtErr := mcmp.ExecAllowAndCompareError(query)
+		// this assumes all queries are valid mysql queries
+		if vtErr != nil {
+			fmt.Println(query)
+			fmt.Println(vtErr)
+			// restart the mysql and vitess connections in case something bad happened
+			closer()
+			mcmp, closer = start(t)
+		}
+		queryCount++
+	}
+	fmt.Printf("Queries successfully executed: %d\n", queryCount)
+}
+
+// TODO: bunch of TestFailingQueries checks were deleted by refactor to use AST
+func randomQuery(schemaTables []tableT, maxAggrs, maxGroupBy int) *sqlparser.Select {
```

Comment:
consider moving all of the generation to a different file, to separate the test running and the query generation a bit